packages/next/client/dev/error-overlay/format-webpack-messages.js

```diff
@@ -171,18 +171,44 @@ function formatMessage(message, verbose) {
 
 function formatWebpackMessages(json, verbose) {
   const formattedErrors = json.errors.map(function (message) {
+    let importTraceNote
+
     // TODO: Shall we use invisible characters in the original error
     // message as meta information?
-    if (
-      message &&
-      message.message &&
-      / (Client|Server) Components compilation\./.test(message.message)
-    ) {
+    if (message && message.message && /NEXT_RSC_ERR_/.test(message.message)) {
       // Comes from the "React Server Components" transform in SWC, always
       // attach the module trace.
+      const NEXT_RSC_ERR_REACT_API = /.+NEXT_RSC_ERR_REACT_API: (.*?)\n/s
+      const NEXT_RSC_ERR_SERVER_IMPORT =
+        /.+NEXT_RSC_ERR_SERVER_IMPORT: (.*?)\n/s
+      const NEXT_RSC_ERR_CLIENT_IMPORT =
+        /.+NEXT_RSC_ERR_CLIENT_IMPORT: (.*?)\n/s
+
+      if (NEXT_RSC_ERR_REACT_API.test(message.message)) {
+        message.message = message.message.replace(
+          NEXT_RSC_ERR_REACT_API,
+          `\n\nYou're importing a component that needs $1. It only works in a Client Component but none of its parents are marked with "client", so they're Server Components by default.\n\n`
+        )
+        importTraceNote =
+          '\n\nMaybe one of these should be marked as a "client" entry:\n'
```

Comment:
"Maybe" seems still bit obscure, could we rephrase it bit like `"One of these could be marked as 'client' entry to allow using client imports"`

Reply:
The problem is that it might not be possible nor desirable to do so. It might be that should actually be using a completely different wrapper component or that you need to break it apart and restructure one of them.